src/org/openvv/js/OVVAsset.js

```diff
@@ -1884,8 +1884,15 @@ Function.prototype.memoize = function() {
         return fn.memoized.apply(fn, arguments);
     }
 };
+var newOVV = new OVV();
 // initialize the OVV object if it doesn't exist
-window.$ovv = window.$ovv || new OVV();
+window.$ovv = window.$ovv || newOVV;
+for(var i in newOVV)
```

Comment:
This is an interim solution until we find a more robust way of dealing with the problem of forward compatibility, which was first brought up in our discussions over a year ago. Assuming that the code is backwards compatible, which it always should be, it's not dangerous. Even if the code isn't backwards compatible, we're only borrowing missing properties, not really merging objects, so the risk is minimal. The fix has been tested in real world scenarios. Without it, measurability drops ~20 points between v1.0 and v1.2 (due to other ads with v1.0 being served on the same page before ads with v1.2). With it we are a few points up from v1.0 to v1.2. We have rigorously tested the solution to make sure the viewability readings are correct as well.

- Amit

On Sep 16, 2015, at 5:29 AM, Tim De Pauw <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/org/openvv/js/OVVAsset.js<https://github.com/InteractiveAdvertisingBureau/openvv/pull/97#discussion_r39609650>:

>  // initialize the OVV object if it doesn't exist
> -window.$ovv = window.$ovv || new OVV();
> +window.$ovv = window.$ovv || newOVV;
> +for(var i in newOVV)


Isn't it really dangerous to just merge these objects, regardless of their version?

—
Reply to this email directly or view it on GitHub<https://github.com/InteractiveAdvertisingBureau/openvv/pull/97/files#r39609650>.